src/WebJobs.Extensions.MobileApps/Config/MobileAppsConfiguration.cs

```diff
@@ -53,8 +62,194 @@ public void Initialize(ExtensionConfigContext context)
                 throw new ArgumentNullException("context");
             }
 
-            context.Config.RegisterBindingExtension(
-                new MobileTableAttributeBindingProvider(context.Config, this, context.Config.NameResolver));
+            INameResolver nameResolver = context.Config.GetService<INameResolver>();
+            IConverterManager converterManager = context.Config.GetService<IConverterManager>();
+
+            BindingFactory factory = new BindingFactory(nameResolver, converterManager);
+
+            IBindingProvider outputProvider = factory.BindToGenericAsyncCollector<MobileTableAttribute, MobileTableContext>(typeof(MobileTableAsyncCollector<>), BindForOutput);
+
+            IBindingProvider clientProvider = factory.BindToExactType<MobileTableAttribute, IMobileServiceClient>(BindForClient);
+
+            IBindingProvider queryProvider = factory.BindToGenericItem<MobileTableAttribute>(BindForQuery);
+            queryProvider = factory.AddTypeFilter(IsQueryType, queryProvider);
+            queryProvider = factory.AddValidator<MobileTableAttribute>(ThrowIfGenericArgumentIsInvalid, queryProvider);
+
+            IBindingProvider jObjectTableProvider = factory.BindToExactType<MobileTableAttribute, IMobileServiceTable>(BindForTable);
+
+            IBindingProvider tableProvider = factory.BindToGenericItem<MobileTableAttribute>(BindForTable);
+            tableProvider = factory.AddTypeFilter(IsTableType, tableProvider);
+            tableProvider = factory.AddValidator<MobileTableAttribute>(ThrowIfGenericArgumentIsInvalid, tableProvider);
+
+            IBindingProvider itemProvider = factory.BindToGenericItem<MobileTableAttribute>(BindForItem);
+            itemProvider = factory.AddValidator<MobileTableAttribute>(IsItemType, itemProvider);
+
+            IExtensionRegistry extensions = context.Config.GetService<IExtensionRegistry>();
+            extensions.RegisterBindingRules<MobileTableAttribute>(ValidateMobileAppUri, nameResolver, outputProvider, clientProvider, jObjectTableProvider, queryProvider, tableProvider, itemProvider);
+        }
+
+        internal bool IsQueryType(Type paramType)
```

Comment:
I'm thinking it would be good to move all this binding stuff out into its own MobileAppsBindingProvider class to seperate concerns. For all of these bindings. Thoughts? That way this config class stays focused on plugging in the extension.